### ISSUE TITLE
Compile with `com.apple.security.get-task-allow` entitlement

### ISSUE THREAD

**Issue by jasonhemann** [railwaycat/homebrew-emacsmacport#262]
I believe right now the pre-built versions are built without this entitlement. As such, users must already have SIP disabled to debug. This is unfortunate when trying to diagnose an intermittent failure.

**Comment by jmhammond** [comment]
I think for your own debugging purposes, you can try to enable that (if this is something you can enable for CLI build tools and not just `xcode`), but for the purposes of the general user, `com.apple.security.get-task-allow` should not be enabled. 
[Apple writes in the documentation](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/resolving_common_notarization_issues)
> This setting tells Xcode to add the com.apple.security.get-task-allow entitlement to your app at build time. This entitlement facilitates debugging on a system that uses System Integrity Protection (SIP) by circumventing certain security checks. However, this poses a security risk for a shipping app, because it can allow an attacker to inject code at runtime.
[And here's a blog](https://wojciechregula.blog/post/dangerous-get-task-allow-entitlement/) where the author shows how to craft an exploit using that flag.

In short, let's not enable the `com.apple.security.get-task-allow` by default.

**Comment by jasonhemann** [comment]
Great answer. Closing.
